docs/tutorials/rerank/rerank_pipeline_with_Bedrock_Rerank_model.md

```diff
@@ -0,0 +1,670 @@
+            },
+            "type": "INLINE"
+        },
+        {
+            "inlineDocumentSource": {
+                "textDocument": {
+                    "text": "Capital punishment (the death penalty) has existed in the United States since beforethe United States was a country. As of 2017, capital punishment is legal in 30 of the 50 states."
+                },
+                "type": "TEXT"
+            },
+            "type": "INLINE"
+        },        
+    ]
+)
+
+results = response["results"]
+print(json.dumps(results, indent=2))
+```
+
+The reranked results are ordered by the highest score:
+```
+[
+  {
+    "index": 2,
+    "relevance_score": 0.7711548724998493
+  },
+  {
+    "index": 0,
+    "relevance_score": 0.0025114635138098534
+  },
+  {
+    "index": 1,
+    "relevance_score": 2.4876490010363496e-05
+  },
+  {
+    "index": 3,
+    "relevance_score": 6.339210403977635e-06
+  }
+]
+```
+
+
+To sort the results by index, use the following code:
+```python
+print(json.dumps(sorted(results, key=lambda x: x['index']),indent=2))
+```
+
+The following are the results sorted by index:
+```
+[
+  {
+    "index": 0,
+    "relevance_score": 0.0025114635138098534
+  },
+  {
+    "index": 1,
+    "relevance_score": 2.4876490010363496e-05
+  },
+  {
+    "index": 2,
+    "relevance_score": 0.7711548724998493
+  },
+  {
+    "index": 3,
+    "relevance_score": 6.339210403977635e-06
+  }
+]
+```
+
+If you see an authorization error when running the test code, check the rerank model access settings on Amazon Bedrock. To check and add model access settings, see [Add or remove access to Amazon Bedrock foundation models](https://docs.aws.amazon.com/bedrock/latest/userguide/model-access-modify.html).
+
+## Step 1: Create a connector and register the model
+
+To create a connector and register the model, use the following steps.
+
+### Step 1.1: Create a connector for the model
+
+First, create a connector for the model.
+
+If you are using self-managed OpenSearch, supply your AWS credentials:
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "Amazon Bedrock Rerank API",
+  "description": "Test connector for Amazon Bedrock Rerank API",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "credential": {
+    "access_key": "your_access_key",
+    "secret_key": "your_secret_key",
+    "session_token": "your_session_token"
+  },
+  "parameters": {
+    "service_name": "bedrock",
+    "endpoint": "bedrock-agent-runtime",
+    "region": "your_bedrock_model_region_like_us-west-2",
+    "api_name": "rerank",
+    "model_id": "amazon.rerank-v1:0"
+  },
+  "actions": [
+    {
+      "action_type": "PREDICT",
+      "method": "POST",
+      "url": "https://${parameters.endpoint}.${parameters.region}.amazonaws.com/${parameters.api_name}",
+      "headers": {
+        "x-amz-content-sha256": "required",
+        "content-type": "application/json"
+      },
+      "pre_process_function": "connector.pre_process.bedrock.rerank",
```

Comment:
I am confused when I see connector.pre_process.bedrock.rerank and different pre-process--function here https://github.com/opensearch-project/ml-commons/blob/654b8a38450a3b07896ba0c19d1a9054bc8a3d1e/docs/tutorials/rerank/rerank_pipeline_with_Amazon_Rerank_model_on_Amazon_Bedrock.md?plain=1#L127-147

Reply:
This blueprint is for for 2.17, before releasing built-in pre/post processor introduced since 2.19.